crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -833,6 +835,87 @@ fn handle_module_level_own_line_comment_before_class_or_function_comment<'a>(
     }
 }
 
+/// Handles the attaching the
+/// ```python
+/// a = "input"[
+///     1 # c
+///     # d
+///     :2
+/// ]
+/// ```
+fn handle_slice_comments<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    let expr_slice = match comment.enclosing_node() {
+        AnyNodeRef::ExprSlice(expr_slice) => expr_slice,
+        AnyNodeRef::ExprSubscript(expr_subscript) => {
+            if expr_subscript.value.end() < expr_subscript.slice.range().start() {
+                if let Expr::Slice(expr_slice) = expr_subscript.slice.as_ref() {
+                    expr_slice
+                } else {
+                    return CommentPlacement::Default(comment);
+                }
+            } else {
+                return CommentPlacement::Default(comment);
+            }
+        }
+        _ => return CommentPlacement::Default(comment),
+    };
+
+    let ExprSlice {
+        range: _,
+        lower,
+        upper,
+        step,
+    } = expr_slice;
+
+    let assignment =
+        assign_comment_in_slice(comment.slice().range(), locator.contents(), expr_slice);
+    let node = match assignment {
+        ExprSliceCommentSection::Lower => lower,
+        ExprSliceCommentSection::Upper => upper,
+        ExprSliceCommentSection::Step => step,
+    };
+
+    if let Some(node) = node {
+        if comment.slice().start() < node.start() {
+            // We can have a comment like
+            // ```
+            // "a"[ # comment
+            //     1:
+            // ]
+            // ```
+            // We format this as an own line comment so we have to change it to one or we'll get
+            // unstable formatting due to leading spaces
+            let comment = if comment.text_position().is_end_of_line() {
+                comment.change_position(CommentTextPosition::OwnLine)
+            } else {
+                comment
+            };
+            CommentPlacement::leading(node.as_ref().into(), comment)
+        } else {
+            // If a trailing comment is an end of line comment that's fine because we have a node
+            // ahead of it
+            CommentPlacement::trailing(node.as_ref().into(), comment)
+        }
+    } else {
+        // We can have a comment like
+        // ```
+        // "a"[ : # comment
+        // ]
+        // ```
+        // We format this as an own line comment so we have to change it to one or we'll get
+        // unstable formatting due to leading spaces
```

Comment:
Is this something that we want to follow up later (add a todo) or are fine with for now (I'm fine with this but we should then document it in our Black divergence document)